File: app/src/main/java/com/originb/inkwisenote2/modules/smartnotes/data/SmartNotebookUpdateType.java
package com.originb.inkwisenote2.modules.smartnotes.data;

public enum SmartNotebookUpdateType {
    NOTE_UPDATE(0),
    NOTE_DELETED(1),
    NOTEBOOK_DELETED(2),
    NOTEBOOK_TITLE_UPDATED(3),;

    private final int updateType;

    SmartNotebookUpdateType(int updateType) {
        this.updateType = updateType;
    }

    public int toInt() {
        return updateType;
    }

    @Override
    public String toString() {
        return String.valueOf(updateType);
    }

    public boolean equals(int updateType) {
        return this.updateType == updateType;
    }
}

File: androidApp/src/main/java/com/originb/inkwisenote2/modules/smartnotes/data/SmartNotebookUpdateType.kt
package com.originb.inkwisenote2.modules.smartnotes.data

enum class SmartNotebookUpdateType(private val updateType: Int) {
    NOTE_UPDATE(0),
    NOTE_DELETED(1),
    NOTEBOOK_DELETED(2),
    NOTEBOOK_TITLE_UPDATED(3),
    ;

    fun toInt(): Int {
        return updateType
    }

    override fun toString(): String {
        return updateType.toString()
    }

    fun equals(updateType: Int): Boolean {
        return this.updateType == updateType
    }
}
